### ISSUE TITLE
"Get events" API endpoint only returns a single result if streaming isn't used

### ISSUE THREAD

**Issue by arctic-alpaca** [containers/podman#10474]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**
Hello,
the [get events API endpoint](https://docs.podman.io/en/latest/_static/api.html#operation/SystemEventsLibpod) only returns a single result if streaming isn't used. If this is intended, it doesn't seem to be documented.

**Steps to reproduce the issue:**

1. Start podman API.

2. Request events with curl: ``curl -X GET --unix-socket /home/`whoami`/testing.sock "http://d/v3.1.2/libpod/events?since=60m&until=1622039711674168800&stream=false"`` 

3. Compare to output of streaming request: ``curl -X GET --unix-socket /home/`whoami`/testing.sock "http://d/v3.1.2/libpod/events?since=60m&until=1622039711674168800"`` 

**Describe the results you received:**
Requests with `stream=0` or `stream=false` result in a single response while the streaming equivalent over the same duration returns a lot more results.

**Describe the results you expected:**
The same amount of results for requests with streaming and without.

**Additional information you deem important (e.g. issue happens only occasionally):**

Output of the requests:
```
alpaca@DESKTOP-3COMHL6:/mnt/c/Users/alpaca/Downloads$ curl -X GET --unix-socket /home/`whoami`/testing.sock "http://d/v3.1.2/libpod/events?since=60m&until=1622039711674168800&stream=false"
{"Type":"image","Action":"pull","Actor":{"ID":"","Attributes":{"containerExitCode":"0","image":"","name":""}},"scope":"local","time":1622038780,"timeNano":1622038780931954200}
```

```
alpaca@DESKTOP-3COMHL6:/mnt/c/Users/alpaca/Downloads$ curl -X GET --unix-socket /home/`whoami`/testing.sock "http://d/v3.1.2/libpod/events?since=60m&until=1622039711674168800"
{"Type":"image","Action":"pull","Actor":{"ID":"","Attributes":{"containerExitCode":"0","image":"","name":""}},"scope":"local","time":1622038780,"timeNano":1622038780931954200}
{"Type":"image","Action":"pull","Actor":{"ID":"","Attributes":{"containerExitCode":"0","image":"","name":""}},"scope":"local","time":1622038836,"timeNano":1622038836606987100}
{"Type":"image","Action":"pull","Actor":{"ID":"","Attributes":{"containerExitCode":"0","image":"","name":""}},"scope":"local","time":1622038851,"timeNano":1622038851054619800}
{"Type":"image","Action":"pull","Actor":{"ID":"","Attributes":{"containerExitCode":"0","image":"","name":""}},"scope":"local","time":1622039057,"timeNano":1622039057057624900}
{"Type":"container","Action":"create","Actor":{"ID":"1f29c5316522ba90077ff5f290867797a9dbcafe8a998d1f99b889471fde9ca9","Attributes":{"containerExitCode":"0","image":"docker.io/library/hello-world:latest","name":"dazzling_snyder"}},"scope":"local","time":1622039337,"timeNano":1622039337796203700}
{"Type":"container","Action":"init","Actor":{"ID":"1f29c5316522ba90077ff5f290867797a9dbcafe8a998d1f99b889471fde9ca9","Attributes":{"containerExitCode":"0","image":"docker.io/library/hello-world:latest","name":"dazzling_snyder"}},"scope":"local","time":1622039338,"timeNano":1622039338004437900}
{"Type":"container","Action":"start","Actor":{"ID":"1f29c5316522ba90077ff5f290867797a9dbcafe8a998d1f99b889471fde9ca9","Attributes":{"containerExitCode":"0","image":"docker.io/library/hello-world:latest","name":"dazzling_snyder"}},"scope":"local","time":1622039338,"timeNano":1622039338010774500}
{"Type":"container","Action":"attach","Actor":{"ID":"1f29c5316522ba90077ff5f290867797a9dbcafe8a998d1f99b889471fde9ca9","Attributes":{"containerExitCode":"0","image":"docker.io/library/hello-world:latest","name":"dazzling_snyder"}},"scope":"local","time":1622039338,"timeNano":1622039338010922400}
{"Type":"container","Action":"died","Actor":{"ID":"1f29c5316522ba90077ff5f290867797a9dbcafe8a998d1f99b889471fde9ca9","Attributes":{"containerExitCode":"0","image":"docker.io/library/hello-world:latest","name":"dazzling_snyder"}},"scope":"local","time":1622039338,"timeNano":1622039338042279900}
{"Type":"container","Action":"remove","Actor":{"ID":"1f29c5316522ba90077ff5f290867797a9dbcafe8a998d1f99b889471fde9ca9","Attributes":{"containerExitCode":"0","image":"docker.io/library/hello-world:latest","name":"dazzling_snyder"}},"scope":"local","time":1622039338,"timeNano":1622039338088990900}
{"Type":"container","Action":"create","Actor":{"ID":"6819ddc71af616e7ed86e4ba1ebd133aa72a4681a43c3506205050c7bf893fae","Attributes":{"containerExitCode":"0","image":"docker.io/library/hello-world:latest","name":"pedantic_feistel"}},"scope":"local","time":1622039711,"timeNano":1622039711393754500}
{"Type":"container","Action":"init","Actor":{"ID":"6819ddc71af616e7ed86e4ba1ebd133aa72a4681a43c3506205050c7bf893fae","Attributes":{"containerExitCode":"0","image":"docker.io/library/hello-world:latest","name":"pedantic_feistel"}},"scope":"local","time":1622039711,"timeNano":1622039711589346900}
{"Type":"container","Action":"start","Actor":{"ID":"6819ddc71af616e7ed86e4ba1ebd133aa72a4681a43c3506205050c7bf893fae","Attributes":{"containerExitCode":"0","image":"docker.io/library/hello-world:latest","name":"pedantic_feistel"}},"scope":"local","time":1622039711,"timeNano":1622039711595883100}
{"Type":"container","Action":"attach","Actor":{"ID":"6819ddc71af616e7ed86e4ba1ebd133aa72a4681a43c3506205050c7bf893fae","Attributes":{"containerExitCode":"0","image":"docker.io/library/hello-world:latest","name":"pedantic_feistel"}},"scope":"local","time":1622039711,"timeNano":1622039711596197300}
{"Type":"container","Action":"died","Actor":{"ID":"6819ddc71af616e7ed86e4ba1ebd133aa72a4681a43c3506205050c7bf893fae","Attributes":{"containerExitCode":"0","image":"docker.io/library/hello-world:latest","name":"pedantic_feistel"}},"scope":"local","time":1622039711,"timeNano":1622039711625742500}
{"Type":"container","Action":"remove","Actor":{"ID":"6819ddc71af616e7ed86e4ba1ebd133aa72a4681a43c3506205050c7bf893fae","Attributes":{"containerExitCode":"0","image":"docker.io/library/hello-world:latest","name":"pedantic_feistel"}},"scope":"local","time":1622039711,"timeNano":1622039711674168800}
```

I'm not sure whether the stream should terminate after the `until` time is reached, it doesn't as far as I can tell.

**Output of `podman version`:**

```
Version:      3.1.2
API Version:  3.1.2
Go Version:   go1.16
Git Commit:   51b8ddbc22cf5b10dd76dd9243924aa66ad7db39-dirty
Built:        Tue Jan  1 01:00:00 1980
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.20.1
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: 'conmon: /usr/libexec/podman/conmon'
    path: /usr/libexec/podman/conmon
    version: 'conmon version 2.0.27, commit: '
  cpus: 12
  distribution:
    distribution: debian
    version: "10"
  eventLogger: file
  hostname: DESKTOP
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.4.72-microsoft-standard-WSL2
  linkmode: static
  memFree: 12298137600
  memTotal: 13346447360
  ociRuntime:
    name: crun
    package: 'crun: /usr/bin/crun'
    path: /usr/bin/crun
    version: |-
      crun version 0.18.1-7931a-dirty
      commit: 7931a1eab0590eff4041c1f74e2844b297c31cea
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /tmp/podman-run-1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    selinuxEnabled: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: 'slirp4netns: /usr/bin/slirp4netns'
    version: |-
      slirp4netns version 1.1.8
      commit: unknown
      libslirp: 4.3.1-git
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.4.4
  swapFree: 4294967296
  swapTotal: 4294967296
  uptime: 5h 1m 3.98s (Approximately 0.21 days)
registries:
  search:
  - docker.io
  - quay.io
store:
  configFile: /home/alpaca/.config/containers/storage.conf
  containerStore:
    number: 4
    paused: 0
    running: 0
    stopped: 4
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: 'fuse-overlayfs: /usr/bin/fuse-overlayfs'
      Version: |-
        fusermount3 version: 3.4.1
        fuse-overlayfs: version 1.4
        FUSE library version 3.4.1
        using FUSE kernel interface version 7.27
  graphRoot: /home/alpaca/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 15
  runRoot: /tmp/podman-run-1000/containers
  volumePath: /home/alpaca/.local/share/containers/storage/volumes
version:
  APIVersion: 3.1.2
  Built: 315532800
  BuiltTime: Tue Jan  1 01:00:00 1980
  GitCommit: 51b8ddbc22cf5b10dd76dd9243924aa66ad7db39-dirty
  GoVersion: go1.16
  OsArch: linux/amd64
  Version: 3.1.2
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

Tested and reproduced with a static build from https://cirrus-ci.com/task/4981016969347072.

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/master/troubleshooting.md)**

Yes, I tested with this static build: https://cirrus-ci.com/task/4981016969347072

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Windows 10, 64bit, WSL2, Debian 10

**Comment by arctic-alpaca** [comment]
> I'm not sure whether the stream should terminate after the until time is reached, it doesn't as far as I can tell.

After some more tests, it does seem to terminate at the appropriate time.

I found some other combinations that don't work as expected:
* `podman events --since=5h --until=1m` and `` curl -X GET --unix-socket /home/`whoami`/testing.sock "http://d/v3.0.0/libpod/events?since=5h&until=1m"`` immediately return without output.  `podman events --since=5h` does return multiple entries.
* `podman events --until=1m` returns one entry: `2021-04-11 19:01:01.2912941 +0200 CEST image pull`, 
``curl -X GET --unix-socket /home/`whoami`/testing.sock "http://d/v3.0.0/libpod/events?until=1m"`` returns the same entry `{"Type":"image","Action":"pull","Actor":{"ID":"","Attributes":{"containerExitCode":"0","image":"","name":""}},"scope":"local","time":1618160461,"timeNano":1618160461291294100}`
* At 18:52 UTC: `podman events --since=2006-01-02T15:04:05 --until=2021-05-26T18:50:55` returns `2021-04-11 19:01:01.2912941 +0200 CEST image pull` while `podman events --since=2006-01-02T15:04:05 --until=2021-05-26T18:54:55` returns the real logs with a huge amount of entries. Same with the API.

Not sure if this is related or a different issue.

**Comment by vrothberg** [comment]
This has been fixed in the main branch by commit 86610c7852ee992d6c0b89d4a5cd70081950951d.  Thanks for opening the issue!
